src/app/components/header/Header.js

```diff
@@ -0,0 +1,107 @@
+"use client"
+
+import React, { useEffect } from "react"
+import AuthModel from "../authmodel/AuthModel"
+import { useState } from "react"
+import { signOut, getCurrentUser } from "aws-amplify/auth"
+
+export default function Header() {
+  const [showModal, setShowModal] = useState(false)
+  const [authStates, setAuthStates] = useState("")
+  const [isSignin, setIsSignin] = useState(false)
+
+  useEffect(() => {
+    async function getUserState(){
+    try{
+      // const start = Date.now()
+      const { signInDetails, username, userId } = await getCurrentUser()
+      // console.log(Date.now() - start)
+      console.log(signInDetails, username, userId )
+
+      setIsSignin(!!userId)
+    }  catch (error) {
+      
+      console.log("error signing out: ", error)
+    }}
```

Comment:
As per [document](https://docs.amplify.aws/react/build-a-backend/auth/connect-your-frontend/manage-user-sessions/#retrieve-your-current-authenticated-user), if a user is not authenticated getCurrentUser would throw an error, instead of returning empty userId, so we should catch that error and set sign in status

```suggestion
      setIsSignin(true)
    }  catch (error) {
      if (error.name === 'UserUnAuthenticatedException') {
      setIsSignin(false)
    }
      console.log("getCurrentUser error: ", error)
    }}
```

[document of UserUnAuthenticatedException](https://docs.amplify.aws/gen1/react/build-a-backend/auth/auth-migration-guide/#authcurrentcredentials-deprecated)